packages/protocol/contracts/libs/LibRealMath.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+library LibRealMath {
+    function exp(uint128 b) internal pure returns (uint256) {
+        // https://github.com/NovakDistributed/macroverse/blob/master/contracts/RealMath.sol
```

Comment:
This one from Remco seems nice: https://github.com/recmo/experiment-solexp/blob/main/src/FixedPointMathLib.sol

I think my previous implementation is probably not good enough because it was only accurate around a single point (but maybe using a lot of rounds would still be okay). I'm not sure what Remco's implementation uses, doesn't really look like Taylor.

Reply:
Using e-curve will be tried on a different PR.